TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TeleOp.java

```diff
@@ -84,29 +79,25 @@ public void runOpMode() {
 
 //------------------Gripper Control-----------------
 
-            //Close the gripper
-            if (gamepad2.right_bumper && robot.gripperState != CLOSED && robot.gripperState != PWMDISABLED)  {
-                gripperPos = robot.gripperOneServo.getPosition() + GRIPPER_MODIFY;
-            }
-
-            //Open the gripper
-            if(gamepad2.right_trigger > 0.5 && robot.gripperState != OPEN && robot.gripperState != PWMDISABLED) {
-                gripperPos = robot.gripperOneServo.getPosition() - GRIPPER_MODIFY;
-            }
-
-            //automatically turn off the servo when grabbing a block
-            if(robot.gripperState == MIDDLE) {
-                robot.gripperOneServo.setPwmEnable();
+            if (robot.isGripperDisabled() == false) {
+                //Move the gripper
+                if (gamepad2.right_bumper)  {
+                    robot.moveGripperClosed();
+                } else if (gamepad2.right_trigger > 0.5) {
+                    robot.moveGripperOpen();
+                } else {
+                    robot.getGripperOneServo().setPwmDisable();
```

Comment:
You're disabling the PWM whenever you aren't pressing the buttons, which is fine (assuming it doesn't affect performance, see the next note), but unnecessary (it's not going to burn out if it's not fighting anything). Are you disabling the PWM when the touch sensors are closed? That's what's really important.

Reply:
In the commit "added a condition", the PWM is only disabled if the gripper state is not MIDDLE (which leaves CLOSED and MAX_OPEN). Also just another note it is important to remember that it needs to be possible to let go of the block once the state reaches CLOSED.

Reply:
OK, but the check added in that commit only applies when no buttons are being pressed. I believe the servo will stay powered until the driver lets go of the button. Can you test that on the bot?

Reply:
I tested it and you are right. The commit "changed conditions to fix TeleOp logic" fixes the issue.